src/hub/HubRegistry.sol

```diff
@@ -34,16 +32,16 @@ contract HubRegistry is Auth, IHubRegistry {
     }
 
     /// @inheritdoc IHubRegistry
-    function registerPool(address manager_, uint16 centrifugeId, AssetId currency_)
+    function registerPool(uint48 poolId_, address manager_, uint16 centrifugeId, AssetId currency_)
         external
         auth
         returns (PoolId poolId)
     {
         require(manager_ != address(0), EmptyAccount());
         require(!currency_.isNull(), EmptyCurrency());
-        require(currency[poolId].isNull(), PoolAlreadyRegistered());
 
-        poolId = newPoolId(centrifugeId, ++latestId);
+        poolId = newPoolId(centrifugeId, poolId_);
```

Comment:
CR 

We should check whether the `PoolId` is well formed instead of adapting it here with the `CentrifugeId`. The UI should just create it and pass it on here IMO.